### PR TITLE
add spellCheck support

### PR DESCRIPTION
**Pull request by terite**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      
No

**Describe what the PR does**    
This PR adds support for passing the `spellCheck` prop to the draft-js Editor component, and enhances the demo to allow passing that prop.

**State whether the PR is ready for review or whether it needs extra work**    

**Additional context**    
* https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/spellcheck
* https://draftjs.org/docs/api-reference-editor#spellcheck

### CODE REVIEW

File: demo/app.js [hunk 1]
```diff
@@ -226,11 +233,21 @@ class App extends React.Component {
               <input
                 id={ 'textIsEditableCheckbox' }
                 type="checkbox"
-                defaultChecked="true"
+                checked={ this.state.isTextEditable }
```
Comment (terite) [inline demo/app.js]: A controlled component seems like the way to go here.

https://reactjs.org/docs/forms.html#handling-multiple-inputs
https://reactjs.org/docs/uncontrolled-components.html#default-values

File: packages/components/timed-text-editor/index.js [hunk 2]
```diff
@@ -509,6 +509,7 @@ class TimedTextEditor extends React.Component {
           blockRendererFn={ this.renderBlockWithTimecodes }
           handleKeyCommand={ command => this.handleKeyCommand(command) }
           keyBindingFn={ e => this.customKeyBindingFn(e) }
+          spellCheck={ this.props.spellCheck }
```
Comment (pietrop) [inline packages/components/timed-text-editor/index.js]:
```suggestion
          spellCheck={ this.props.spellCheck?  this.props.spellCheck : false  }
```

What do you think about having a default to false if the prop is not specified to making it an optional parameter to the component?

**Comment by pietrop** [comment]
Just noticed it works in Chrome, doesn't seem to work in Firefox? (on a mac)

**Comment by terite** [comment]
I'm able to get spell checking on macOS 10.14.4 using Firefox `66.0.5 (64-bit)`


Does FireFox have a dictionary pre-installed for your locale? https://support.mozilla.org/en-US/kb/how-do-i-use-firefox-spell-checker

In any case, actual spell checking is browser dependent and out of my control as per links in the description.

File: packages/components/timed-text-editor/index.js (hunk 2 above)
Comment (terite) [reply packages/components/timed-text-editor/index.js]: This parameter is already optional. `this.props.spellCheck` will evaluate to `true`, `false`, or `undefined`. All 3 are valid values. Additionally, the relevant prop type is not suffixed with `isRequired`.

**Pull request merged**